doc/build/sysbuild/index.rst

```diff
@@ -0,0 +1,501 @@
+         :compact:
+
+This builds ``hello_world`` and ``mcuboot`` for the ``reel_board``, and then
+flashes both the ``mcuboot`` and ``hello_world`` application images to the
+board.
+
+More detailed information regarding the use of MCUboot with Zephyr can be found
+in the `MCUboot with Zephyr`_ documentation page on the MCUboot website.
+
+.. note::
+
+   MCUBoot default configuration will perform a full chip erase when flashed.
+   This can be controlled through the MCUBoot Kconfig option
+   ``CONFIG_ZEPHYR_TRY_MASS_ERASE``. If this option is enabled, then flashing
+   only MCUBoot, for example using ``west flash --domain mcuboot``, may erase
+   the entire flash, including the main application image.
+
+Sysbuild Kconfig file
+*********************
+
+You can set sysbuild's Kconfig options for a single application using
+configuration files. By default, sysbuild looks for a configuration file named
+``sysbuild.conf`` in the application top-level directory.
+
+In the following example, there is a :file:`sysbuild.conf` file that enables building and flashing with
+MCUboot whenever sysbuild is used:
+
+.. code-block:: none
+
+   <home>/application
+   ├── CMakeLists.txt
+   ├── prj.conf
+   └── sysbuild.conf
+
+
+.. code-block:: none
+
+   SB_CONFIG_BOOTLOADER_MCUBOOT=y
+
+You can set a configuration file to use with the
+``-DSB_CONF_FILE=<sysbuild-conf-file>`` CMake build setting.
+
+For example, you can create ``sysbuild-mcuboot.conf`` and then
+specify this file when building with sysbuild, as follows:
+
+.. tabs::
+
+   .. group-tab:: ``west build``
+
+      .. zephyr-app-commands::
+         :tool: west
+         :app: samples/hello_world
+         :board: reel_board
+         :goals: build
+         :west-args: --sysbuild
+         :gen-args: -DSB_CONF_FILE=sysbuild-mcuboot.conf
+         :compact:
+
+   .. group-tab:: ``cmake``
+
+      .. zephyr-app-commands::
+         :tool: cmake
+         :app: share/sysbuild
+         :board: reel_board
+         :goals: build
+         :gen-args: -DAPP_DIR=samples/hello_world -DSB_CONF_FILE=sysbuild-mcuboot.conf
+         :compact:
+
+.. _sysbuild_zephyr_application:
+
+Adding Zephyr applications to sysbuild
+**************************************
+
+You can use the ``ExternalZephyrProject_Add()`` function to add Zephyr
+applications as sysbuild domains. Call this CMake function from your main
+:file:`CMakeLists.txt` file, or any other CMake file you know will run.
+
+Targeting the same board
+========================
+
+To include ``my_sample`` as another sysbuild domain, targeting the same board
+as the main image, use this example:
+
+.. code-block:: cmake
+
+   ExternalZephyrProject_Add(
+     APPLICATION my_sample
+     SOURCE_DIR <path-to>/my_sample
+     BUILD_ALWAYS True
+   )
+
+This could be useful, for example, if your board requires you to build and flash an
+SoC-specific bootloader along with your main application.
+
+Targeting a different board
```

Comment:
I think that the most common case would be to target a different core or CPU rather than targeting a different board.
Maybe, from sysbuild's perspective core and CPU are just specific cases of "board". If this is the case, from my reader's perspective, it would help the understanding to read it.

Reply:
yes, in Zephyr all targets are defined as boards.
For example for a `nrf5340dk` development kit with two core, those cores can be addressed as:
`nrf5340dk_nrf5340_cpunet` and `nrf5340dk_nrf5340_cpuapp`.

And to even make it more challenging, when building a nonsecure app and using Trustzone (for example by enabling TF-M), then you even has `nrf5340dk_nrf5340_cpuapp_ns`.

And if you look at the example, it is targeting `cpu1`.
I think we should keep referring to board, as that is what is passed through the `BOARD` variable, but I will add a sentence that each core or cpu on a devkit is targeted as BOARD in Zephyr build system.

Reply:
@erwango please find updated description here: https://github.com/zephyrproject-rtos/zephyr/compare/ebc0d4ad715399d360d40857fec185763eb1701d..08034b26f59f4d16227e457e8f04f7510f41e8de

Reply:
"In Zephyr a board may refer to:"
This is always the case, in device tree for instance, boards and socs are different notions.
Can you replace by "For SysBuild", or "In Zephyr build system" ?